### ISSUE TITLE
Question/Error fixing

### ISSUE THREAD

**Issue by vnageshbio** [fhalab/evSeq#37]
Hi,
Thank you for developing this!
I have been trying to give this a try on some of our data and I keep getting this error:
```
Unmatched Files in Folder:


Error Encountered:
Unhandled exception encountered: ''Index''
```

I had to change the index on the first row in index_map.csv based on the indexes we use, so am not sure if that is what is causing the error.

**Comment by vnageshbio** [comment]
Hi @brucejwittmann,
I have been tinkering with the code and trying to get it to work with the sequences we have.
There is a place in the code where it seems to be pulling out the barcode from the fastq itself, but our fastqs dont have that information.
```
    def calculate_read_stats(self, record):
        """Calculate summary stats for a read."""
        # Calculate relevant summary stats needed by both forward and reverse methods
        barcode = str(record.seq)[:BARCODE_LENGTH]
        length = len(record)
        average_quality = np.mean(record.letter_annotations["phred_quality"])
        
        return barcode, length, average_quality
```
Because of this, what happens is that, it doesnt find the well id in the merged data frames between well and index and returns no results.
Can you please suggest a work around?

**Comment by palmhjell** [comment]
Hi @Vaishn17. I'm not entirely sure how to address this, as the primary thing that evSeq accomplishes is extracting barcode pairs from each entry in the fastq files (prepared according to the library prep protocol) and mapping that entry's sequence to its plate-well position. Without the barcodes in the provided fastq files, the evSeq software shouldn't really be doing anything.

Does that make sense? If I'm misunderstanding something please let me know!

**Comment by vnageshbio** [comment]
Hi @palmhjell . Yes I get that. But during the demuxing process, the barcodes go into header or are actually given as an I1.fastq.gz file. Is there a particular way in which the libraries have to be demuxed then?

**Comment by palmhjell** [comment]
You mean the demultiplexing process after the initial multiplexed NGS run? The evSeq-specific barcodes should still be in the sequences.

The evSeq software provides a demultiplexing process that is separate from (and applied after) the initial demultiplexing process of a multiplexed NGS run. Demultiplexing the NGS run should follow a standard protocol based on the provided barcodes. Then evSeq further demultiplexes that sample (which corresponded to a single previous barcode) to map the sequences to plate-well coordinates.

Basically, the NGS is done in a way that is completely agnostic to the underlying evSeq machinery. Once that is done, the fastq file (which still has the evSeq-specific barcodes, because nothing before evSeq knew about them) is submitted to the evSeq software.

**Comment by vnageshbio** [comment]
Oh I see! So the index_map.csv is not illumina specifici I7/i5 adapters? I generated an index_map file which was filled with the i7/i5 adapters.

**Comment by palmhjell** [comment]
No, they were designed and validated by us for the evSeq protocol specifically, and are used in tandem with the i7/i5 adapters (attached after the evSeq full protocol is performed).

I would review [panel B in our workflow](https://github.com/fhalab/evSeq#the-evseq-workflow), which shows how the i7/i5 adaptors are added later, and/or read our manuscript if you haven't, which you can find at [ACS Synthetic Biology](https://pubs.acs.org/doi/10.1021/acssynbio.1c00592) if you have access or [bioRxiv](https://www.biorxiv.org/content/10.1101/2021.11.18.469179v1) if you don't (they are pretty much identical). It explains the rationale behind the protocol and how it can (and we usually do) use i7/i5 adapters _downstream_ of the evSeq procotol.

**Comment by vnageshbio** [comment]
Thank you Patrick! You just cleared the fog for me :) I have been trying to analyze the data and I failed to understand why the wells were not matching up. It makes sense now!
